### PR TITLE
billing project security checks

### PR DESCRIPTION
**Pull request by dvoet**
https://broadinstitute.atlassian.net/browse/DSDEEPB-1948

in addition to what is described in the JIRA, add service account as bucket and default object acl owner
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [ ] **LR**: Initial review by LR and others.
- [ ] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Squash commits, rebase if necessary
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Merge to develop 
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: Verify swagger UI on dev server still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala [hunk 1]
```diff
@@ -131,14 +133,27 @@ class HttpGoogleServicesDAO(
...
       retry(when500) {
         () => Future {
-          val bucketAcls = groupAccessLevelsAscending.map(ac => newBucketAccessControl(makeGroupEntityString(toGroupId(bucketName, ac)), ac))
+          // bucket ACLs should be:
+          //   workspace owner - bucket writer
+          //   workspace writer - bucket writer
+          //   workspace reader - bucket reader
+          //   bucket service account - bucket owner
+          val workspaceAccessToBucketAcl = Map(Owner -> "WRITER", Write -> "WRITER", Read -> "READER")
+          val bucketAcls =
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala:142]: Format `bucketACLs` and `defaultObjectAcls` lines in the same way, to make their similarity more obvious on first glance

File: src/test/scala/org/broadinstitute/dsde/rawls/webservice/EntityApiServiceSpec.scala [hunk 2]
```diff
@@ -455,7 +455,7 @@ class EntityApiServiceSpec extends FlatSpec with HttpService with ScalatestRoute
 
   val attributeList = AttributeValueList(Seq(AttributeString("a"), AttributeString("b"), AttributeBoolean(true)))
   val z1 = Entity("z1", "Sample", Map("foo" -> AttributeString("x"), "bar" -> AttributeNumber(3), "splat" -> attributeList))
-  val workspace2Name = new WorkspaceName(testData.wsName.namespace + "2", testData.wsName.name + "2")
+  val workspace2Name = new WorkspaceName(testData.wsName.namespace, testData.wsName.name + "2")
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/rawls/webservice/EntityApiServiceSpec.scala:458]: why this change?

**Comment by jmthibault79** [comment]
done review.  Only a few comments.

File: src/test/scala/org/broadinstitute/dsde/rawls/webservice/EntityApiServiceSpec.scala (hunk 2 above)
Comment (dvoet) [inline src/test/scala/org/broadinstitute/dsde/rawls/webservice/EntityApiServiceSpec.scala:458]: so I don't need to create another billing project

**Comment by jmthibault79** [comment]
:+1:
